firmware/arduino/chessboard/uart.ino

```diff
@@ -0,0 +1,84 @@
+char incomingByte;
+char  buffer[5];
+int byteNum = -1; //-1 indicates that the start code hasn't been received
+
+//Wait for input
+void serialEvent2()
+{
+    //Loop through all available bytes
+    while (Serial2.available())
+    {
+        //Get byte
+        incomingByte = (char) Serial2.read();
+
+        //Reset buffer position
+        if (incomingByte == '~')
+        {
+            byteNum = 0;
+        }
+        //Add byte to buffer
+        else if (byteNum != -1)
+        {
+            buffer[byteNum++] = incomingByte;
+        }
+
+        //Check if the buffer is full, process input
+        if (byteNum == 5)
+        {
+            //Reset buffer position
+            byteNum = -1;
+
+            //Process input
+            //Returns true for valid input and false for invalid input. Calls movement function
+            parse_message_from_pi(buffer);
+        }
+    }
+}
+
+bool parse_message_from_pi(char * message)
+{
+    //Move types 0 and 3 are valid for the same range of values
+    if (message[0] == 0 || message[0] == 3)
+    {
+        //Check that the inputs are in a valid range
+        if (message[1] < 'a' || message[1] > 'h' || message[2] < '0' ||  message[2] > '8' || message[3] < 'a' || message[3] > 'h' || message[4] < '0' ||  message[4] > '8') 
```

Comment:
I think here you want to check if the values at 2 and 4 are in range [1,8] inclusive rather than [0, 8]. Same comment below

Reply:
Good catch. Thanks!